frontend/src/js/controllers/challengeCtrl.js

```diff
@@ -1415,6 +1415,88 @@
             }
         };
 
+        vm.challengePhaseDialog = function(ev, phase) {
+            vm.page.challenge_phase = phase;
+            vm.phaseStartDate = new Date(phase.start_date);
+            vm.phaseEndDate = new Date(phase.end_date);
+            vm.testAnnotationFile = null;
+            vm.sanityCheckPass = true;
+            vm.sanityCheck = "";
+            $mdDialog.show({
+                scope: $scope,
+                preserveScope: true,
+                targetEvent: ev,
+                templateUrl: 'dist/views/web/challenge/edit-challenge/edit-challenge-phase.html',
+                escapeToClose: false
+            });
+        };
+
+        vm.sanityCheckInit = function() {
+            vm.sanityCheckPass = false;
+        };
+
+        vm.sanityCheckSwitch = function() {
+            vm.sanityCheckPass = true;
+        };
+
+        vm.editChallengePhase = function(editChallengePhaseForm) {
+            if(editChallengePhaseForm){
+                vm.page.challenge_phase.start_date = vm.phaseStartDate;
+                vm.page.challenge_phase.end_date = vm.phaseEndDate;
+                vm.challengePhaseId = vm.page.challenge_phase.id;
+                var parameters = {};
+                parameters.url = "challenges/challenge/" + vm.challengeId + "/challenge_phase/" + vm.challengePhaseId;
+                parameters.method = 'PATCH';
```

Comment:
Now, create a `FormData` object here and pass the values to it one by one, like
`var formData = new FormData();
formData.append("name", vm.page.challenge_phase.name);
formData.append("description", vm.page.challenge_phase.description);` and so on.

Finally after taking all the fields, please assign this to parameters using
`parameters.data = formData`

Please try this and I think it will work. Let me know if ypou still have some doubts or queries.